### ISSUE TITLE
Events Overlapping

### ISSUE THREAD

**Issue by Sys-Admin889** [nextcloud/calendar#2470]
### Expected behaviour
In case of sending two calendar invitations to the same person at the same time, calendar should decline the second one automatically.

### Actual behaviour
 Calendar don't decline events overlapping automatically.

### Calendar app
**Calendar app version:** (2.0.3)

### Client configuration
**Browser:** (Version 84.0.4147.125 (Official Build) (64-bit))

**Operating system:** (Widows 10, Ubuntu 18.04 and Mac OS 10.14)

### Server configuration

**Operating system**: (Univention Corporate Server)

**Database:** (PostgreSQL 9.6.17 on x86_64-pc-linux-gnu)

**PHP version:** (7.2.24)

**Nextcloud Version:** (18.0.6)

**Comment by georgehrke** [comment]
I disagree that this should be done automatically, it should be up to the user to decide what action to take, but the calendar should definitely show a warning if the calendar-user is already booked / when receiving an invitation and overlapping with another event in your calendar.

**Comment by torvalstrom** [comment]
I built this against 6.5.3 and have it running, so here is a concrete proposal rather than a +1.

## What is missing, precisely

The free/busy view already exists, but it lives in the **full** editor behind *Find a time*. The editor people actually use for quick bookings is the popover (`EditSimple.vue`), and there it is not reachable at all - worse, the popover is drawn on top of the grid, so it covers the very calendar that would have shown the clash. The result is that avoiding a double booking in the quick flow depends on remembering your own week.

## The logic already exists

`Service/Appointments/EventConflictFilter.php` already defines what "busy" means for this app: query the user's calendars over the slot, then drop anything with `STATUS:CANCELLED` or `TRANSP:TRANSPARENT`. Booking pages use it; the editor does not. Reusing those exact rules means "busy" keeps meaning one thing across the app, which seems more defensible than inventing a second definition in the frontend.

## What I have working

A small mixin plus a `NcNoteCard` under the date/time fields. It debounces 400 ms, queries each enabled own calendar with `calendar.dav.findByTypeInTimeRange('VEVENT', ...)`, applies the same three exclusions (own UID, cancelled, transparent), lists up to three conflicts with their times and summarises the rest, and offers a button that opens the full editor's free/busy view. It never blocks saving - deliberate double bookings are a real thing. All-day events are skipped, since they overlap everything by definition.

Two things that cost me time and are worth writing down for anyone else who tries this:

1. **`startDate` / `endDate` in the editor are wall-clock dates**, not instants. Passing them straight to a CalDAV time-range query searches the wrong moment as soon as the browser timezone differs from the calendar timezone - in my case nine hours off, so it silently found nothing. Reading `calendarObjectInstance.eventComponent.startDate.getInTimezone(utc).jsDate` gives the real instant.
2. **Watching `startDate`/`endDate` does not fire.** The editor fills them in after the component is created and a watcher on the Date objects never sees it. Watching a primitive derived from them (`${start.getTime()}-${end.getTime()}`) with `immediate: true` works.

## Open questions before I send a PR

- Popover only, or the full editor too?
- Which calendars count - all enabled own calendars, or a configurable subset like the Appointments `calendarFreebusyUris`?
- Should the conflict check move to the backend and reuse `EventConflictFilter` directly, rather than duplicating the rules in the frontend? That is cleaner but a bigger change.

Happy to open the PR if there is interest, and happy to drop it if you would rather solve this by surfacing the existing free/busy view instead.

**Comment by torvalstrom** [comment]
Correction to point 1 of my previous comment - the advice I gave there is right about the problem and wrong about the cure, and the wrong version has a bug I then hit in testing.

Reading the instant from `calendarObjectInstance.eventComponent.startDate.getInTimezone(utc).jsDate` does give the correct moment, but those calendar-js values are **mutated in place**. A computed property that depends on them never invalidates, so the warning keeps showing the answer for the old date after the user picks a new one - it looked completely correct until you changed the date and nothing happened.

What works is to stay on the reactive editor state and convert there: take `startDate` / `endDate` (whose *local* getters carry the intended wall-clock digits) together with `startTimezoneId`, and resolve the pair to an instant with `Intl.DateTimeFormat(..., { timeZone })`, applying the offset twice so a guess landing on the wrong side of a DST switch still settles. The computed then depends only on reactive state and updates as the user edits.

Verified both directions through the real date picker: opening on a busy slot shows the warning and moving to a free day clears it; opening on a free day shows nothing and moving onto the busy day makes it appear.
